ONSdigital/design-system
# Autosuggest - Refactor component and update docs

### What is the context of this PR?
Issues with existing code:
- For non address autosuggest  and non-editable address autosuggest the requirement was to use the `onsInput` macro which would call `onsAutoggest` (so the actual component was hidden). You would need to create an `autosuggest` object to do this. Bit untidy and confusing.
- For editable address autosuggest you would need to use the `onsAddressInput` macro.
- For non-editable autosuggest you would need to manually add `<div>` elements and the client side validation wouldn't work.
- As components were hidden the macro_options table was too big and confusing as it had all `input` component options + address options + autosuggest options.
- The non address autosuggest implementation is completely different to address autosuggest implentation and only shared the same core code and so the documentation was messy.

New way:
- When using the non address autosuggest you now use the `onsAutosuggest` macro. Clearer implementation.
- A new component has been created `onsAddressFinder` for both editable and non-editable address autosuggests (this calls `onsAutosuggest`).
- Existing `onsAddressInput` has been removed.
- Documentation and macro options are now contextual to each component as they're now seperated and specific to each use case.
- Using `isEditable: boolean` is now a required parameter as they use the same macro.
- Added more validation rules for non-editable address.

### How to review
- You can check the documentation/structure etc in the netlify preview.
- Review both autosuggest and address finder components docs etc
- Review the "Addresses" pattern

The actual components won't work in Netlify as there is no JWT supplied. Firstly you'll need your IP whitelisted by the AIMS team. Checkout this branch and do a find and replace for `'APIDomainBearerToken': "some_token",` and replace `some_token` with the JWT that I can send you (over email). You can then start the service and use the components. Then:

- Check validation
- Check things work as before
